resource/airflow_dag_pipeline.yaml

```diff
@@ -49,6 +49,222 @@ jobs:
+              if [ -d "tests" ]; then
+                echo "Running tests in tests dir"
+                docker run --entrypoint "" dag/test:latest pytest tests
+              fi
+      on_failure:
+        put: pull-request
+        attempts: 3
+        params:
+          path: pull-request
+          status: failure
+          context: unittests
+    - task: run-syntax
+      privileged: true
+      config:
+        platform: linux
+        image_resource:
+          type: docker-image
+          source:
+            repository: amidos/dcind
+        inputs:
+          - name: image
+        run:
+          path: sh
+          args:
+            - -ec
+            - |
+              source /docker-lib.sh
+              start_docker
+              docker load -i image/image.tar
+              # Try importing files in base dir to check for syntax errors
+              docker run --entrypoint "" dag/test:latest python -c "
+              import os
+              import sys
+              from importlib import import_module
+              prefix = os.getcwd()
+              print(f'Working directory: {prefix}')
+              for dirpath, dirnames, filenames in os.walk(prefix):
+                  trimmed_mods = [f[:f.find('.py')] for f in filenames if not f.startswith('__') and f.find('.py') > 0]
+                  for m in trimmed_mods:
+                      if dirpath not in sys.path:
+                          sys.path.insert(0, dirpath)
+                      mod = dirpath.replace(prefix + '/', '').replace('/', '.') + '.' + m
+                      print(f'importing: {mod}')
+                      import_module(mod)
+              "
+      on_failure:
+        put: pull-request
+        attempts: 3
+        params:
+          path: pull-request
+          status: failure
+          context: syntax
+    # mark test as passed
+    - put: pull-request
+      attempts: 3
+      params:
+        path: pull-request
+        status: success
+        context: iam-role
+    - put: pull-request
+      attempts: 3
+      params:
+        path: pull-request
+        status: success
+        context: unittests
+    - put: pull-request
+      attempts: 3
+      params:
+        path: pull-request
+        status: success
+        context: syntax
+  - aggregate:
+    - task: delete-iam-role-name
+      config:
+        platform: linux
+        image_resource:
+          type: docker-image
+          source:
+            repository: governmentpaas/awscli
+        inputs:
+          - name: pull-request-edit
+        params:
+          AWS_ACCESS_KEY_ID: ((secrets.role-putter-access-key-id))
+          AWS_SECRET_ACCESS_KEY: ((secrets.role-putter-secret-access-key))
+          AWS_NODES_ROLE_ARN: ((secrets.nodes-role-arn))
+        run:
+          path: sh
+          args:
+            - -ecx
+            - |
+              export ROLE_NAME=$(cat source/deploy.json | jq -r '.role_name')
```

Comment:
You may just hardcode this to `test_dag_role` as that doesn't change.

Also, as discussed offline, there may be a race condition if this test run more than once.
In that case the 2nd+ time the role is deleted aws cli would return an error and this fail (I don't know if there is a way to catch this `404 NOT FOUND`). But maybe it will not happen that often to care at this stage as you also mentioned 👍